Enabled to export mesh as 3dxml format

KenMat765 opened the pull request:
This PR allows the URDF exporter to output mesh files not only in STL format but also in 3DXML format. This enables the export of models with color information.
This does not destroy the original functions.

This solves this issue.
https://github.com/ros/solidworks_urdf_exporter/issues/114

![image](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/c133615c-1a05-42a7-875f-7f21f0bacfb3)

### Exported in STL (no color)
![grayscale](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/07448d93-5d90-4e24-b363-74a947b258b4)

### Exported in 3dxml (with color)
![color](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/e4d43ee8-396e-44c8-b795-a7161e0b1345)

## How to convert to .dae format for ros
 
```
pip3 install scikit-robot -U
convert-urdf-mesh <URDF_PATH> --output <OUTPUT_URDF_PATH>
```

brawner commented on SW2URDF/UI/AssemblyExportForm.cs:
```diff
@@ -307,7 +307,10 @@ private void FinishExport(bool exportSTL)
-                Exporter.ExportRobot(exportSTL);
+
+                bool isSTL = radioButtonStl.Checked;
+                Exporter.ExportRobot(exportSTL, isSTL);
```
Can you make the mesh format an enum instead of a bool? This will help make it easier to add support for more file formats in the future

brawner commented on SW2URDF/URDFExport/ExportHelper.cs:
```diff
@@ -295,18 +296,75 @@ private void ExportFiles(Link link, URDFPackage package, int count, bool exportS
-            string windowsMeshFileName = package.WindowsMeshesDirectory + linkName + ".STL";
+            string meshFilename = package.MeshesDirectory + linkName;
+            string windowsMeshFileName = package.WindowsMeshesDirectory + linkName;
+            if (isSTL)
```
Instead of an if/else, can you use a switch with the enum?

brawner commented on SW2URDF/URDFExport/ExportHelper.cs:
```diff
@@ -295,18 +296,75 @@ private void ExportFiles(Link link, URDFPackage package, int count, bool exportS
             if (exportSTL)
             {
-                SaveSTL(link, windowsMeshFileName);
+                if (isSTL)
```
Use switch

brawner reviewed (commented):
Thank you so much for your PR. I think it would make a great addition to this tool, so my comments are mostly around future maintainability. 

Can you please add:
 * A unit test for this functionality so that when I create a release I know it works appropriately.
 * An additional section in the README to explain converting meshes to DAE

KenMat765 commented:
I'm sorry for the late reply.

I've changed the bool to an enum of the export format.
I've also added a section in the README to explain converting meshes to DAE

Although, I had a problem with the unit test.

## Problem
When I exported  [ORIGINAL_3_DOF_ARM](https://github.com/ros/solidworks_urdf_exporter/blob/master/examples/ORIGINAL_3_DOF_ARM/ORIGINAL_3_DOF_ARM.SLDASM), the links of the arm were shown in the wrong position in Rviz.
This might be related to these issues: https://github.com/ros/solidworks_urdf_exporter/issues/130, https://github.com/ros/solidworks_urdf_exporter/issues/116
I tried with the released code, but it still didn't work.

## Temporary solution
When I created the same coordinates and axis of each link at the top-assembly, and selected those in the export menu instead of those in the sub-assemblies, the links were shown in the correct position.
This worked on both STL and 3DXML formats.

As discussed in https://github.com/ros/solidworks_urdf_exporter/issues/116, I believe this issue can occur with SolidWorks later than 2018, but is this supposed to function correctly originally?

| wrong position in STL | correct position in STL | correct position in 3DXML |
|---|---|---|
| ![ORIGINAL_3_DOF_ARM_STL](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/0fb1bc08-37e9-4e12-a413-e84816c71727) | ![ORIGINAL_3_DOF_ARM_STL_TOPASSEM](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/464d47b0-c750-497c-99b7-8490e1a25e96) | ![ORIGINAL_3_DOF_ARM_3DXML_TOPASSEM](https://github.com/ros/solidworks_urdf_exporter/assets/79335866/28204d0f-2755-475a-a7c4-6543bc409251) |

## Environment
Solidworks version: 2022
urdf_exporter: [1.6.1](https://github.com/ros/solidworks_urdf_exporter/releases/tag/1.6.1)

iory commented:
@brawner 
I believe this PR could be highly beneficial for all SolidWorks and ROS users. I would appreciate it if you could kindly review it.

brawner commented:
@iory I believe this PR is still waiting for a unit test

KenMat765 commented:
I have modified all the points you mentioned:

- Converted bool to enum / if to switch
- Added instructions for converting to .dae in the README
- Added unit tests

I would greatly appreciate it if you could kindly review the changes.

brawner reviewed (approved):
Excellent, thank you for the contribution!
